x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/function/scalar/BinaryScalarFunction.java

```diff
@@ -46,7 +46,7 @@ public final BinaryScalarFunction replaceChildren(List<Expression> newChildren)
         Expression newLeft = newChildren.get(0);
         Expression newRight = newChildren.get(1);
 
-        return left.equals(newLeft) && right.equals(newRight) ? this : replaceChildren(newLeft, newRight);
+        return replaceChildren(newLeft, newRight);
```

Comment:
I'm not sure what to think about this, I may try to do... something around it, as the test code currently needs `replaceChildren()` to actually replace them even if only the Source changed.
This is the only ESQL expression I found doing this, so I need to check if this was made for some specific reason

Reply:
Finally replaced with that, and restored the original code here